### ISSUE TITLE
Including JSON files in project might make compilation fail

### ISSUE THREAD

**Issue by per-samuelsson** [Starcounter/Starcounter#3075]
This one was found by @jwillmer.

If you add a common directory that include random JSON files to a Starcounter project, compilation will fail because Starcounter will consider these to be "JSON-by-example" and they are in fact not.

Its easily reproducable by taking `/Launcher/wwwroot` and do "Include in project".

We've discussed it, and @chrhol is aware of it. We have come up with an idea how to solve it, with the help of @miyconst.

**Comment by chrhol** [comment]
The extension enhances all files with an ending of `.json` with properties that allow runtime generation of code to enable intellisense, and groups it with a codebehind file (`.json.cs`) if it exists

**The problem**: When you have json-files you want to include in a project that should NOT be treated as a TypedJson json-by-example you have to manually remove set properties on EACH file, which is not doable if you have a lot of json-files.

**The solution**: The extension should not enhance single json files (i.e. that have no codebehind file) that are included in a project by default. Instead we should have some option to mark a json file in a project as TypedJson. One possible solution, if not to complex to implement, is to enhance the context menu in the solution explorer in Visual Studio, and have an option to set properties for TypedJson on a json file.

**Comment by chrhol** [comment]
Happened to @joozek78 as well.

**Comment by joozek78** [comment]
Can't we just make it so 'TypedJSON' is not default type for jsons? I always feel that visual studio context menu has already been 'enhanced' to its limits :|

**Comment by chrhol** [comment]
> Can't we just make it so 'TypedJSON' is not default type for jsons? I always feel that visual studio context menu has already been 'enhanced' to its limits :|

I guess it would depend on what the most common usage of json is in a C# Starcounter application. It's not only the type that needs to be set, also the custom tool property needs to be set correctly.

But what about the suggestion to only set these values when you include the code-behind file? Fairly specific to SC TypedJSON and would allow you to include any other json without getting any extra properties set. 

The context menu is more of a sugaring on-top and not really needed. We can document manual steps to set type and custom tool.

**Comment by joozek78** [comment]
> But what about the suggestion to only set these values when you include the code-behind file? Fairly specific to SC TypedJSON and would allow you to include any other json without getting any extra properties set.

This is actually a great idea and I agree with it 100%

**Comment by chrhol** [comment]
I've pushed the change above now. Existing json-files without a code-behind that are added/included to a SC project will no longer be treated as TypedJSON. 

It's still possible to add a code-behind file as well though, and the json and code-behind will be grouped and properties set.

If a single json-file should be TypedJSON, you either have to create a new TypedJSON from the add new item dialog or manually set two properties on the json file:
- Build Action = TypedJSON
- Custom Tool = MSBuild:Compile

I've pushed the change to develop and soon pnext.

**NOTE**: The change is in the extension so make sure that the extension is reinstalled to make it work.
